marc_to_solr/lib/solr_deleter.rb

```diff
@@ -0,0 +1,44 @@
+require "net/http"
+require "json"
+
+# This class is to allow us to delete Solr records with our current
+# version of Traject. Once we upgrade to Traject version 3.1 or later
+# we could use Traject's native delete feature
+# (see https://github.com/traject/traject/commit/c731d09c6cb15fbe572e4a294a7a127b525b7277)
+# but keep in mind that this feature is still in experimental mode in Traject
+# and their implementation does not support batches.
+class SolrDeleter
+  def initialize(solr_url)
+    @solr_url = solr_url
+  end
+
+  def delete(ids, batch_size = 100)
+    ids.each_slice(batch_size) do |batch|
+      response = delete_batch(batch)
+      raise StandardError, "Error deleting Solr documents. Status: #{response.code}. Body: #{response.body}" if response.code != "200"
```

Comment:
I think we'll want the list of ids to get logged, as well.

I also have some questions here which I think might make sense for a follow-up ticket.

1. In other places where we do batch operations, we retry one-at-a-time on failures like this. Perhaps we should log the error like this, then recurse with a smaller batch size (maybe go straight to batch_size 1)? 
1. What reaction is required if a record or a batch of records fails to get deleted? I.e. is some human intervention required? It seems likely that it would be, so this might be a good place for a honeybadger notification so that we are alerted to look into it.

Reply:
I've added logging using Traject's native logger (if available). I am not sure if/how this would connect to HoneyBadger but at least deletes would be logged as any other Traject operation.

Reply:
If we want to try failed deletes with smaller batches I would agree that we should create a separate ticket.